python/cugraph/cugraph/dask/link_analysis/pagerank.py

```diff
@@ -13,31 +13,41 @@
 # limitations under the License.
 #
 
+import warnings
+
+import dask
 from dask.distributed import wait, default_client
-import cugraph.dask.comms.comms as Comms
 import dask_cudf
 import cudf
 import numpy as np
-import warnings
-from cugraph.dask.common.input_utils import get_distributed_data
-
 from pylibcugraph import (
+    pagerank as plc_pagerank,
+    personalized_pagerank as plc_p_pagerank,
+    exceptions as plc_exceptions,
     ResourceHandle,
-    pagerank as pylibcugraph_pagerank,
-    personalized_pagerank as pylibcugraph_p_pagerank,
 )
 
+import cugraph.dask.comms.comms as Comms
+from cugraph.dask.common.input_utils import get_distributed_data
+from cugraph.exceptions import FailedToConvergeError
+
 
-def convert_to_cudf(cp_arrays):
+def convert_to_return_tuple(cp_arrays):
     """
     Creates a cudf DataFrame from cupy arrays from pylibcugraph wrapper
     """
-    cupy_vertices, cupy_pagerank = cp_arrays
+    cupy_vertices, cupy_pagerank, *converged = cp_arrays
+
     df = cudf.DataFrame()
     df["vertex"] = cupy_vertices
     df["pagerank"] = cupy_pagerank
 
-    return df
+    if len(converged) > 0:
+        converged = converged[0]
```

Comment:
The `converged` list here must be length 1, right? I would make this unpacking more explicit by e.g.
```python
[converged] = converged
```
(if you like this style).

Reply:
Thanks, I refactored this in [88156bf](https://github.com/rapidsai/cugraph/pull/3639/commits/88156bf2b0eed402fd8e4b0d2c66881a4b591dc5)